lib/component/catalog_table.es6.js

```diff
@@ -59,6 +53,34 @@ function catalogTableController($scope, api) {
     const releaseKeys = getReleaseKeys(releases);
     matrixController.setReleaseKeys(releaseKeys);
     controller.releaseKeys = releaseKeys;
+
+    // If this is the only matrixController call, then wait for all updates
+    // to arrive.
+    controller.waitState = 0;
+  };
+  const setOpts = function(opts) {
+    matrixController.setOpts(opts);
+
+    // If this is the only matrixController call, then wait for "DATA" update.
+    controller.waitState = controller.waitState & (~WAIT_FLAGS.DATA);
+  };
+  const setReleasesAndOpts = function(releases, opts) {
+    setReleases(releases);
+    const releasesWaitState = controller.waitState;
+    setOpts(opts);
+    const optsWaitState = controller.waitState;
+
+    // Wait for most conservative set of updates determined by individual calls.
```

Comment:
The reason I have the flags and this complex update is that I have a sneaking suspicion that worker API dependencies on data fetch and data rehashing may change. It would be nice if multiple-call helpers like this one don't need updating if/when they do.